frontend/ui/screensaver.lua

```diff
@@ -383,6 +384,34 @@ function Screensaver:setMessage()
     self.input_dialog:onShowKeyboard()
 end
 
+function Screensaver:setStretchLimit(touchmenu_instance)
+    UIManager:show(SpinWidget:new{
+        value = G_reader_settings:readSetting("screensaver_stretch_limit_percentage", 8),
+        value_min = 0,
+        value_max = 25,
+        default_value = 8, -- percent
+        title_text = _("Set maximum stretch limit"),
+        ok_text = _("Set"),
+        ok_always_enabled = true,
+        callback = function(spin)
+            G_reader_settings:saveSetting("screensaver_stretch_limit_percentage", spin.value)
+            G_reader_settings:makeTrue("screensaver_stretch_images")
+            if touchmenu_instance then touchmenu_instance:updateItems() end
+        end,
+        extra_text = _("Disable stretch"),
+        extra_callback = function()
+            G_reader_settings:makeFalse("screensaver_stretch_images")
+            if touchmenu_instance then touchmenu_instance:updateItems() end
+        end,
+        option_text = _("Always stretch"),
```

Comment:
Ambiguous wording; you mean "enable stretch, without any limit", dunno how to make that shorter.
But "Always stretch" feels like the other won't be "always".

Reply:
Full stretch?